nixos/modules/services/cluster/k3s/default.nix

```
@@ -147,8 +147,8 @@ in
 
     systemd.services.k3s = {
       description = "k3s service";
-      after = [ "network.service" "firewall.service" ];
-      wants = [ "network.service" "firewall.service" ];
+      after = [ "network-online.target" "firewall.service" ];
+      wants = [ "firewall.service" ];
```

```suggestion
      wants = [ "firewall.service" "network-online.target" ];
```

`man systemd.special`

> Units that strictly require a configured network connection should pull in network-online.target (via a Wants= type dependency) and order themselves after it.

---

Done. Please review.